packages/zoltan2/src/algorithms/order/Zoltan2_AlgAMD.hpp

```diff
@@ -146,6 +149,10 @@ class AlgAMD : public Algorithm<Adapter>
       // wgts are ignored in AMD
       model->getEdgeList(edgeIds, offsets, wgts);
 
+#if AMD_WITH_LNO_T
+    {
+
+      // SR: We will always call AMD with lno_t
```

Comment:
I realized I didn't correctly answer your second question. If long and long long are both 64 bit, this won't be a problem.  I assume you are asking what if Suitesparse_long was 32 bits. In this case, we will (might ?) get a error from Teuchos::as. I don't think any of the tests used lno_t as long long.